mlir/CMakeLists.txt

```diff
@@ -84,6 +84,7 @@ set(MLIR_MIOPEN_DRIVER_MISC_E2E_TEST_ENABLED 0 CACHE BOOL "Enable build miscella
 set(MLIR_MIOPEN_DRIVER_PR_E2E_TEST_ENABLED 0 CACHE BOOL "Enable build PR-triggered E2E tests for MIOpen driver")
 set(MLIR_MIOPEN_DRIVER_RANDOM_DATA_SEED "none" CACHE STRING "Enable E2E tests using random data")
 set(MLIR_MIOPEN_DRIVER_XDLOPS_TEST_ENABLED 0 CACHE BOOL "Enable build E2E tests for XDLops")
+set(MLIR_MIOPEN_DRIVER_GPU_VALIDATION_ENABLED 0 CACHE BOOL "Enable E2E tests with GPU validation")
```

Comment:
Just a general note: this sounds like it enables/disables GPU validation at a code level, not just for the tests. Maybe rename this?

Also, shouldn't it be on by default?

Reply:
MLIR_MIOPEN_DRIVER_TEST_WITH_GPU_VALIDATION? MLIR_MIOPEN_DRIVER_GPU_VALIDATION_TEST_ENABLED? Can you suggestion one? I'm not good at making up names.

I want it to be off by default. By default we should do -pv which means verifying with C++ kernels, as -pv is more reliable.

Reply:
Makes sense on it being off by default.

I'd think `MLIR_MIOPEN_DRIVER_TEST_GPU_VALIDATION` could work